ddl/ddl.go

```diff
@@ -207,6 +210,17 @@ func NewDDL(store kv.Storage, infoHandle *infoschema.Handle, hook Callback, leas
 	return newDDL(store, infoHandle, hook, lease)
 }
 
+// TODO: Move this to the function of newDDL.
+func (d *ddl) setWorker(ctx goctx.Context, cli *clientv3.Client) {
+	d.worker = &worker{
+		ddlID:      d.uuid,
+		etcdClient: cli,
+	}
+	ctx, d.worker.cancelFunc = goctx.WithCancel(ctx)
+	d.wait.Add(2)
+	go d.campaignOwners(ctx)
```

Comment:
`d.wait.Add(2)`  looks strange here, it can be changed:

```
d.campaignOwners(ctx)

func (d *ddl) campaignOwners(ctx goctx.Context) {
 	d.worker.newSession(ctx, newSessionDefaultRetryCnt)
    d.wait.Add(2)
 	go d.campaignLoop(ctx, ddlOwnerKey)
 	go d.campaignLoop(ctx, bgOwnerKey)
 }
```